src/repos/UserRepo.ts

```diff
@@ -50,13 +42,18 @@ const deleteUser = async (user: User): Promise<boolean> => {
 
 const updateUser = async (
   user: User,
-  firstName: string,
-  lastName: string,
-  netID: string
+  userFields: UserUpdateFields
 ): Promise<boolean> => {
-  user.firstName = firstName ? firstName : user.firstName;
-  user.lastName = lastName ? lastName : user.lastName;
-  user.netID = netID ? netID : user.netID;
+  const userFieldKeys = Object.keys(userFields);
+  if (userFieldKeys.length < 1) {
+    throw Error('At least one user field is required');
+  }
+  for (const key of userFieldKeys) {
+    if (!Object.keys(user).includes(key)) {
+      throw Error('Invalid user field provided: ' + key);
```

Comment:
what if they provide all of the proper user fields but added in something extra..

Reply:
I had it working that way initially but I felt like it could cause a really hard to track down bug on iOS/Android because if they make a typo when sending the request they would still get a `{success: true}` response even though the user didn't fully update.